keras_nlp/samplers/top_p_sampler.py

```diff
@@ -34,6 +34,8 @@ class TopPSampler(Sampler):
 
     Args:
         p: float, the `p` value of top-p.
+        sample_from_top_k: int, defaults to None. If set, only sample from top
```

Comment:
I would just call this `k` honestly (parallel with the same option in top-k), and in the description  that it is a heuristic cutoff point.

Reply:
My thought is having both `p` and `k` is a bit confusing to readers, here `p` is the main arg while `k` is a supporting role, using `p` and `k` combo kinda suggest they are parallel to each other.

No strong opinion here and agree `sample_from_top_k` isn't a great name.